platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/model/internal/StandardWiringModel.java

```diff
@@ -255,11 +218,58 @@ public void registerEdge(
         }
     }
 
+    /**
+     * Register an input wire with the wiring model. For every input wire registered via this method, the model expects
+     * to see exactly one registration via {@link #registerInputWireBinding(String, String)}.
+     *
+     * @param taskSchedulerName the name of the task scheduler that the input wire is associated with
+     * @param inputWireName     the name of the input wire
+     */
+    public void registerInputWireCreation(
+            @NonNull final String taskSchedulerName, @NonNull final String inputWireName) {
+        final boolean unique = inputWires.add(new InputWireDescriptor(taskSchedulerName, inputWireName));
+        if (!unique) {
+            throw new IllegalStateException(
+                    "Duplicate input wire " + inputWireName + " for scheduler " + taskSchedulerName);
+        }
+    }
+
+    /**
+     * Register an input wire binding with the wiring model. For every input wire registered via
+     * {@link #registerInputWireCreation(String, String)}, the model expects to see exactly one registration via this
+     * method.
+     *
+     * @param taskSchedulerName the name of the task scheduler that the input wire is associated with
+     * @param inputWireName     the name of the input wire
+     */
+    public void registerInputWireBinding(@NonNull final String taskSchedulerName, @NonNull final String inputWireName) {
+        final InputWireDescriptor descriptor = new InputWireDescriptor(taskSchedulerName, inputWireName);
+
+        final boolean registerd = inputWires.contains(descriptor);
+        if (!registerd) {
+            throw new IllegalStateException(
+                    "Input wire " + inputWireName + " for scheduler " + taskSchedulerName + " was not registered");
+        }
+
+        final boolean unique = boundInputWires.add(descriptor);
+        if (!unique) {
+            throw new IllegalStateException("Input wire " + inputWireName + " for scheduler " + taskSchedulerName
+                    + " should not be bound more than once");
+        }
+    }
+
     /**
      * {@inheritDoc}
      */
     @Override
     public void start() {
+
+        // We don't have to do anything with the output of these sanity checks.
+        // The methods below will log errors if they find problems.
+        checkForCyclicalBackpressure();
+        checkForIllegalDirectSchedulerUsage();
+        checkForUnboundInputWires();
```

Comment:
Attn @alittley: you can remove these checks from `PlatformWiring.java`. I didn't make that change in this PR in order to reduce merge conflicts.